internal/tss/channels/error_jitter_channel.go

```diff
@@ -0,0 +1,97 @@
+package channels
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	"github.com/alitto/pond"
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/wallet-backend/internal/tss"
+	"github.com/stellar/wallet-backend/internal/tss/router"
+	"github.com/stellar/wallet-backend/internal/tss/services"
+	"golang.org/x/exp/rand"
+)
+
+type ErrorJitterChannelConfigs struct {
+	TxManager            services.TransactionManager
+	Router               router.Router
+	MaxBufferSize        int
+	MaxWorkers           int
+	MaxRetries           int
+	MinWaitBtwnRetriesMS int
+}
+
+type errorJitterPool struct {
+	Pool                 *pond.WorkerPool
+	TxManager            services.TransactionManager
+	Router               router.Router
+	MaxRetries           int
+	MinWaitBtwnRetriesMS int
+}
+
+var ErrorJitterChannelName = "ErrorJitterChannel"
+
+var _ tss.Channel = (*errorJitterPool)(nil)
+
+func jitter(dur time.Duration) time.Duration {
+	halfDur := int64(dur / 2)
+	delta := rand.Int63n(halfDur) - halfDur/2
+	return dur + time.Duration(delta)
+}
+
+func NewErrorJitterChannel(cfg ErrorJitterChannelConfigs) *errorJitterPool {
+	pool := pond.New(cfg.MaxBufferSize, cfg.MaxWorkers, pond.Strategy(pond.Balanced()))
+	return &errorJitterPool{
+		Pool:                 pool,
+		TxManager:            cfg.TxManager,
+		Router:               cfg.Router,
+		MaxRetries:           cfg.MaxRetries,
+		MinWaitBtwnRetriesMS: cfg.MinWaitBtwnRetriesMS,
+	}
+}
+
+func (p *errorJitterPool) Send(payload tss.Payload) {
+	p.Pool.Submit(func() {
+		p.Receive(payload)
+	})
+}
+
+func (p *errorJitterPool) Receive(payload tss.Payload) {
+	ctx := context.Background()
+	var i int
+	for i = 0; i < p.MaxRetries; i++ {
+		currentBackoff := p.MinWaitBtwnRetriesMS * (1 << i)
+		time.Sleep(jitter(time.Duration(currentBackoff)) * time.Microsecond)
+		rpcSendResp, err := p.TxManager.BuildAndSubmitTransaction(ctx, ErrorJitterChannelName, payload)
+		if err != nil {
+			log.Errorf("%s: unable to sign and submit transaction: %e", ErrorJitterChannelName, err)
+			return
+		}
+		payload.RpcSubmitTxResponse = rpcSendResp
+		if !slices.Contains(tss.JitterErrorCodes, rpcSendResp.Code.TxResultCode) {
+			err = p.Router.Route(payload)
+			if err != nil {
+				log.Errorf("%s: unable to route payload: %e", ErrorJitterChannelName, err)
+				return
+			}
+			return
+		}
+	}
+	if i == p.MaxRetries {
+		// Retry limit reached, route the payload to the router so it can re-route it to this pool and keep re-trying
+		// NOTE: Is this a good idea? Infinite tries per transaction ?
```

Comment:
Wdyt about sending an error webhook request in this case instead of the infinite loop?
Given that there is the `max retries` env var as a client I wouldn't expect this to be retried indefinitely.

Reply:
So max_retries is really a variable for how many retries we want in a single task - since we do not want each task submitted to the go routine pool to run infinitely. But in theory, each transaction can be retried an infinite number of times. In this case, like the comment points out, the payload will get routed to the same go routine pool, but it will just be a brand new task. 
I'm allowing for infinite retries on transactions because that's the way SDP does it. If we find that this is becoming an issue in testing, we can cap it

Reply:
Got it. Does SDP also have this concept of "max retries"?

Reply:
No, that is specific to this implementation. SDP retries transactions infinitely. They essentially have an infinite loop which picks up transactions that need to be processed from a table

Reply:
Do you think it would make sense for the error jitter and non jitter channels to simply sleep for the required time (based on the records of # of tries in the DB) and route the payload to the rpc caller service?
That way we could achieve the same, but would have the build and submission of TX's in a central place/pool.
And then the sleep time is always exponential (it never resets when it reached `max tries`.

Reply:
But the jitter and non jitter channel dont wait the same way. There is no/constant wait between retries in the non jitter channel and an exponential backoff in the jitter channel. And its not based on the # or records in the tries table - although that could be done that way, but i feel like that would be a little less clean